src/views/clusteroverview/ClusterOverviewPage.tsx

```diff
@@ -38,6 +44,17 @@ const ClusterOverviewPage: React.FC = () => {
     },
   ];
 
+  const showModal = useMemo(
+    () => !quickStarts?.dontShowWelcomeModal,
+    [quickStarts?.dontShowWelcomeModal],
+  );
+
+  useEffect(() => {
+    showModal &&
+      createModal(({ isOpen, onClose }) => <WelcomeModal isOpen={isOpen} onClose={onClose} />);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []); // we do need to render this only once
```

Comment:
Adding the deps here is causing the modal to spawn more than one?
What happens if you do something like:
```
useEffect(() => {
  !quickStarts?.dontShowWelcomeModal && createModal....
}, [quickStarts?.dontShowWelcomeModal, createModal]);
```
and remove the useMemo above?

Reply:
It happens that the modal won't occur, but the blank screen, and your browser gets stuck. I recommend you to try it. This is why I also added the comment there, to prevent someone accidentally changing that and adding deps to the array.